### ISSUE TITLE
Avoiding process leakage in web request.

### ISSUE THREAD

**Issue by justgage** [Voronchuk/elixir_google_spreadsheets#43]
The way we're trying to use this library is by appending to a Google Sheet in the process of a web request. However when you do `GSS.Spreadsheet.Supervisor.spreadsheet(sheet_id)` it will create a process under the `GSS.Spreadsheet.Supervisor` supervision tree. These don't get cleaned up so those processes never stop running which causes a "process leak". 

You can see them pooling up here:
<img width="783" alt="image" src="https://user-images.githubusercontent.com/164033/78921296-efa92200-7a51-11ea-95f6-2b3fa660df64.png">

The workaround for us is to do this:
```elixir
with {:ok, pid} <- GSS.Spreadsheet.Supervisor.spreadsheet(sheet_id),
  # do stuff to it...
  :ok = Supervisor.terminate_child(GSS.Spreadsheet.Supervisor, pid) do
:ok
end
```

I feel like the design of the library kind if encourages you to do this.

**Comment by Voronchuk** [comment]
@justgage thanks for the notice, it's definitely smth which should be addressed, as a process can be re-used from the registry. I'll try to take a look this weekend but you can suggest a PR and I'll look at it for sure.

**Comment by Voronchuk** [comment]
Fixed in 0.1.17
